### PR TITLE
CR-1079709: Add an option to generate all profiling files in a sub direcotory

### PR DESCRIPTION
**Pull request by jvillarre**
This pull request adds the option "profiling_directory" to the xrt.ini file which specifies a directory to dump all of the generated profiling files, including AIE profiling, power profiling, summary file, and trace files.  The run summary will still be generated in the current working directory.  

This initial submission only adds Linux support.  If the directory doesn't exist or the application cannot create the directory, then files will still be dumped in the working directory.

### CODE REVIEW

File: src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp [hunk 1]
```diff
@@ -28,10 +36,61 @@ namespace xdp {
...
+      fout.open(filename) ;
+      return ;
+    }
+
+    // The directory was specified.  Check if it exists and is writable
+    bool dirExists  = false ;
+    bool writeable  = false ;
+#ifdef _WIN32
+#else
+    struct stat buf ;
```
Comment (stsoe) [inline src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp:62]: warning: uninitialized record type: `buf` [cppcoreguidelines-pro-type-member-init]

```cpp
struct stat buf ;
^
               {}
```

Comment (stsoe) [inline src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp:62]: warning: uninitialized record type: `buf` [hicpp-member-init]

```cpp
struct stat buf ;
^
               {}
```

File: src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp [hunk 2]
```diff
@@ -28,10 +36,61 @@ namespace xdp {
...
+        // The directory exists, but can we write to it?
+        writeable = (buf.st_mode & S_IWUSR) != 0 ;
+      }
+      //else {
+        // The file exists, but it is not a directory.  We cannot write to it
+      //}
+    }
+    else {
+      // The file does not exist at all, so try to create it
+      result = mkdir(directory.c_str(), 0777) ;
```
Comment (stsoe) [inline src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp:77]: warning: 0777 is a magic number; consider replacing it with a named constant [cppcoreguidelines-avoid-magic-numbers]

```cpp
result = mkdir(directory.c_str(), 0777) ;
                                  ^
```

**Review by stsoe — COMMENTED** [review]
clang-tidy made some suggestions

File: src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp (hunk 2 above)
Comment (jvillarre) [reply src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp]: Added named constant rwx_all for the 0777 value in latest commit.

File: src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp (hunk 1 above)
Comment (jvillarre) [reply src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp]: Added initialization

**Review by IshitaGhosh — APPROVED** [review]
Changes look good,

**Review by hackwa — APPROVED** [review]
looks good. we might need windows implementation too in future

**Review by pgschuey — APPROVED** [review]
looks good

**Pull request merged**